lib/rspec/core/shared_example_group.rb

```diff
@@ -33,10 +33,31 @@ def include_in(klass, inclusion_line, args, customization_block)
         klass.update_inherited_metadata(@metadata) unless @metadata.empty?
 
         SharedExampleGroupInclusionStackFrame.with_frame(@description, inclusion_line) do
-          klass.class_exec(*args, &@definition)
+          klass_exec(klass, *args, &@definition)
           klass.class_exec(&customization_block) if customization_block
         end
       end
+
+    private
+
+      if RSpec::Support::RubyFeatures.kw_args_supported?
+        # Remove this in RSpec 4 in favour of explictly passed in kwargs down the entire
+        # stack, e.g. rspec/rspec-core#2711
+        binding.eval(<<-CODE, __FILE__, __LINE__)
+        def klass_exec(klass, *args, &definition)
+          if RSpec::Support::MethodSignature.new(definition).has_kw_args_in?(args)
+            kwargs = args.pop
+            klass.class_exec(klass, *args, **kwargs, &definition)
```

Comment:
That's strange, do we have to pass `klass` as a first argument?

Reply:
er we do not, thats a mistake